spec/helper/versioning_helper_spec.rb

```diff
@@ -794,6 +860,282 @@ def mock_native_releases
+      ),
+      AppendPhcVersionIfNecessaryParams.new(
+        interactive: false,
+        append_on_confirmation: false,
+        append_hybrid_common_version: nil,
+        include_prereleases: false,
+        hybrid_common_version: " ",
+        new_version_number: "2.0.0",
+        expected_version: "2.0.0"
+      ),
+      AppendPhcVersionIfNecessaryParams.new(
+        interactive: false,
+        append_on_confirmation: false,
+        append_hybrid_common_version: nil,
+        include_prereleases: false,
+        hybrid_common_version: nil,
+        new_version_number: "2.0.0",
+        expected_version: "2.0.0"
+      ),
+      AppendPhcVersionIfNecessaryParams.new(
+        interactive: false,
+        append_on_confirmation: false,
+        append_hybrid_common_version: nil,
+        include_prereleases: false,
+        hybrid_common_version: "12.0.0",
+        new_version_number: " ",
+        expected_version: " "
+      ),
+      AppendPhcVersionIfNecessaryParams.new(
+        interactive: false,
+        append_on_confirmation: false,
+        append_hybrid_common_version: nil,
+        include_prereleases: false,
+        hybrid_common_version: "12.0.0",
+        new_version_number: nil,
+        expected_version: nil
+      ),
+      AppendPhcVersionIfNecessaryParams.new(
+        interactive: false,
+        append_on_confirmation: false,
+        append_hybrid_common_version: nil,
+        include_prereleases: false,
+        hybrid_common_version: "12.0.0",
+        new_version_number: "2.0.0-SNAPSHOT",
+        expected_version: "2.0.0-SNAPSHOT"
+      ),
+      AppendPhcVersionIfNecessaryParams.new(
+        interactive: false,
+        append_on_confirmation: false,
+        append_hybrid_common_version: nil,
+        include_prereleases: false,
+        hybrid_common_version: "12.0.0",
+        new_version_number: "2.0.0-alpha.1",
+        expected_version: "2.0.0-alpha.1"
+      ),
+      AppendPhcVersionIfNecessaryParams.new(
+        interactive: false,
+        append_on_confirmation: false,
+        append_hybrid_common_version: nil,
+        include_prereleases: false,
+        hybrid_common_version: "12.0.0",
+        new_version_number: "2.0.0-beta.1",
+        expected_version: "2.0.0-beta.1"
+      ),
+      AppendPhcVersionIfNecessaryParams.new(
+        interactive: false,
+        append_on_confirmation: false,
+        append_hybrid_common_version: nil,
+        include_prereleases: false,
+        hybrid_common_version: "12.0.0",
+        new_version_number: "2.0.0+meta",
+        expected_version: "2.0.0+meta"
+      ),
+      AppendPhcVersionIfNecessaryParams.new(
+        interactive: false,
+        append_on_confirmation: false,
+        append_hybrid_common_version: nil,
+        include_prereleases: false,
+        hybrid_common_version: "12.0.0",
+        new_version_number: "2.0.0+",
+        expected_version: "2.0.0+"
+      )
+    ]
+
+    params.each_with_index do |param, i|
```

Comment:
Nitpick but wdyt about calling the `params` array something like `params_sets` and then calling `param` `params` instead?

Reply:
Yea I like it! https://github.com/RevenueCat/fastlane-plugin-revenuecat_internal/pull/67/commits/1a729a73806028dfa996a1ec54769966819a6838